apps/api-server/src/routes/widget/widget.js

```diff
@@ -38,26 +38,64 @@ router
 
     return next();
   })
-  .get((req, res, next) => {
+  .get(async (req, res, next) => {
+    const projectId = req.query.projectId;
     const widgetId = Math.floor(Math.random() * 1000000);
     const randomId = Math.floor(Math.random() * 1000000);
     const componentId = `osc-component-${widgetId}-${randomId}`;
     const widgetType = req.widgetConfig.widgetType;
-    const widgetSettings = widgetSettingsMapping[widgetType];
+    let widgetSettings = widgetSettingsMapping[widgetType];
+
+    if (!widgetSettings) {
+      return next(
+        createError(400, 'Invalid widget type given for fetching settings')
+      );
+    }
 
     // Remove widgetType from config, but pass all other keys to the widget
     delete req.widgetConfig.widgetType;
-    const widgetConfig = JSON.stringify(req.widgetConfig);
+    let projectConfig = {};
+    let defaultConfig = {};
 
-    let output = getWidgetJavascriptOutput(
-      widgetSettings,
-      widgetType,
-      componentId,
-      widgetConfig
-    );
+    if (projectId) {
+      try {
+        const project = await db.Project.findOne({
+          where: {
+            id: projectId,
+          },
+        });
+
+        if (project) {
+          projectConfig = project.config;
+        } else {
+          createError(404, 'Could not find the project belonging to given id');
+        }
+        defaultConfig = getDefaultConfig(projectId);
+      } catch (e) {
+        return next(createError(500, 'Could not fetch the project'));
+      }
+    }
+
+    try {
+      const output = setConfigsToOutput(
+        widgetType,
+        componentId,
+        widgetSettings,
+        defaultConfig,
+        projectConfig,
+        flattenObject(req.widgetConfig)
```

Comment:
Dit werkt niet. Zo is er in de config
```
  "api": {
    "url": "https://api.os20.nlsvgtr.nl"
  },
  "login": {
    "url": "https://api.os20.nlsvgtr.nl/auth/project/2/login?useAuth=default&redirectUri=[[REDIRECT_URI]]"
  },
  "logout": {
    "url": "https://api.os20.nlsvgtr.nl/auth/project/2/logout?useAuth=default&redirectUri=[[REDIRECT_URI]]"
  },
```
en daar blijft dus alleen de tweede van over, met als veldnaam 'url'.

Reply:
Hoe bedoel je dit? Ik verwijder niet de een key uit de config van wat ik nu zie, dus dat zou geen issue moeten zijn

Reply:
Dit zou nu opgelost moeten zijn door geen gelaagdheid meer mee te geven in de widget config. De `flattenObject` is er nu uitgehaald.